### PR TITLE
Add CompositionEvent bindings

### PR DESCRIPTION
**Pull request by pyfisch**
Needed for better keyboard input. See also #17578

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18093)

### CODE REVIEW

**Comment by jdm** [comment]
```
./components/script/dom/compositionevent.rs:64: no newline at EOF
```

**Comment by pyfisch** [comment]
Fix newline and configure editor to always insert newline at end of file.

@bors-servo: try

**Comment by pyfisch** [comment]
> assert_true: If this is failing: DANGER, are you sure you want to expose the new interface CompositionEvent to all webpages as a property on the global? Do not make a change to this file without review from jdm or Ms2ger for that specific change! expected true got false

@jdm, @Ms2ger do you agree to expose the new interface?

**Comment by jdm** [comment]
Yep.

**Comment by jdm** [comment]
One thing to be aware of - I would hesitate to merge this PR as-is, since we don't have any code that uses the CompositionEvent interface. If websites use it for feature-detection, they would get incorrect results.

**Comment by pyfisch** [comment]
Feature-detection is a good point. Should I feature gate composition events or do you propose something else? I try to propose my changes as small, self-contained PRs but they not always use all of the new code immediately.

**Comment by pyfisch** [comment]
@bors-servo try

Out of curiosity, why is `./mach update-manifest` needed?

**Comment by jdm** [comment]
MANIFEST.json contains hashes of the test files, unfortunately. I would accept a PR that hides the CompositionEvent interface behind a `[Pref="dom.compositionevent.enabled"]` annotation; this would require reverting the test-related changes.

File: components/script/dom/webidls/CompositionEvent.webidl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*
+ * The origin of this IDL file is
+ * https://w3c.github.io/uievents/#idl-compositionevent
+ *
+ */
+
+// https://w3c.github.io/uievents/#idl-compositionevent
+[Constructor(DOMString type, optional CompositionEventInit eventInitDict)]
+interface CompositionEvent : UIEvent {
```
Comment (jdm) [inline components/script/dom/webidls/CompositionEvent.webidl]: Please add a `Pref="dom.compositionevent.enabled"` annotation and add a new preference to resources/prefs.json.

File: tests/wpt/mozilla/tests/mozilla/interfaces.html [hunk 2]
```diff
@@ -19,6 +19,7 @@
   "CanvasPattern",
   "CharacterData",
   "CloseEvent",
+  "CompositionEvent",
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/interfaces.html]: All of the changes under tests/ can be reverted.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by pyfisch** [comment]
requested changes done

**Pull request merged**